lib/Reconciler.lua

```diff
@@ -306,6 +312,22 @@ function Reconciler._reconcileInternal(instanceHandle, newElement)
 		return nil
 	end
 
+	-- Check for wasted renders if the config is set
+	if GlobalConfig.getValue("trackWastedRenders") then
+		local type
+		if isPrimitiveElement(newElement) then
+			type = "Primitive"
+		elseif isFunctionalElement(newElement) then
+			type = "Functional"
+		elseif isStatefulElement(newElement) then
+			type = "Stateful"
+		else
```

Comment:
This seems like it'd break if we add another kind of element in the future. Could we change this `else` to `elseif isPortal(newElement)`, and warn (error?) if we couldn't determine the type?